server/services/deadlineService.js

```diff
@@ -0,0 +1,47 @@
+const logger = require('../../log.js')
+const { getRoOverdueCasesDate, getRoDueCasesDates } = require('../utils/dueDates')
+
+const roleStages = {
+  RO: 'PROCESSING_RO',
+}
+
+const dueDateMethods = {
+  RO: { due: getRoDueCasesDates, overdue: getRoOverdueCasesDate },
+}
+
+module.exports = function createDeadlineService(licenceClient) {
+  async function getDueInDays(role, days) {
+    if (!Number.isInteger(days) || days < 0) {
+      throw new Error('Days must be a whole number')
+    }
+    rejectIfUnknown(role)
+    const { from, upto } = dueDateMethods[role].due(days)
+    try {
```

Comment:
Curious.. Is there any need to wrap this in a try catch simply so you can log an error..
The method `getDueInDays ` and  erroring method `getLicencesInStageBetweenDates` is well named so it would be easy to find and debug from the logs?

Reply:
No, there isn't really. I was just being consistent with most of the other service code which does the same thing, also without any real need to. I'll take it out here. Probably nees to come out in lots of other old code too, eventually.